velox/functions/lib/tests/SliceTest.cpp

```diff
@@ -13,117 +13,169 @@
+      testSlice("slice(C0, -6, 5)", {arrayVector}, expectedArrayVector);
+    }
+    {
+      auto arrayVector = makeArrayVector<int64_t>({{1, 2, 3, 4}});
+      auto expectedArrayVector = makeArrayVector<int64_t>({{}});
+      testSlice("slice(C0, -6, 5)", {arrayVector}, expectedArrayVector);
+    }
+    {
+      // The implementation is zero-copy, so floating point number comparison
+      // won't have issue here since numerical representation remains the same.
+      auto arrayVector = makeArrayVector<double>({{2.3, 2.3, 2.2}});
+      auto expectedArrayVector = makeArrayVector<double>({{2.3, 2.2}});
+      testSlice("slice(C0, 2, 3)", {arrayVector}, expectedArrayVector);
+    }
+    // String array.
+    {
+      auto arrayVector = makeArrayVector<StringView>({{"a", "b", "c", "d"}});
+      auto expectedArrayVector = makeArrayVector<StringView>({{"b", "c"}});
+      testSlice("slice(C0, 2, 2)", {arrayVector}, expectedArrayVector);
+    }
+    // Out of bound start index.
+    {
+      auto arrayVector = makeArrayVector<StringView>({{"a", "b", "c", "d"}});
+      auto expectedArrayVector = makeArrayVector<StringView>({{}});
+      testSlice("slice(C0, 5, 2)", {arrayVector}, expectedArrayVector);
+    }
+    // Out of bound length.
+    {
+      auto arrayVector = makeArrayVector<StringView>({{"a", "b", "c", "d"}});
+      auto expectedArrayVector = makeArrayVector<StringView>({{"b", "c", "d"}});
+      testSlice("slice(C0, 2, 5)", {arrayVector}, expectedArrayVector);
+    }
   }
-  {
-    auto arrayVector = makeArrayVector<int64_t>({{1, 2, 3, 4}});
-    auto expectedArrayVector = makeArrayVector<int64_t>({{}});
-    testSlice("slice(C0, -5, 5)", {arrayVector}, expectedArrayVector);
-  }
-  {
-    auto arrayVector = makeArrayVector<int64_t>({{1, 2, 3, 4}});
-    auto expectedArrayVector = makeArrayVector<int64_t>({{}});
-    testSlice("slice(C0, -6, 5)", {arrayVector}, expectedArrayVector);
-  }
-  {
-    auto arrayVector = makeArrayVector<int64_t>({{1, 2, 3, 4}});
-    auto expectedArrayVector = makeArrayVector<int64_t>({{}});
-    testSlice("slice(C0, -6, 5)", {arrayVector}, expectedArrayVector);
-  }
-  {
-    // The implementation is zero-copy, so floating point number comparison
-    // won't have issue here since numerical representation remains the same.
-    auto arrayVector = makeArrayVector<double>({{2.3, 2.3, 2.2}});
-    auto expectedArrayVector = makeArrayVector<double>({{2.3, 2.2}});
-    testSlice("slice(C0, 2, 3)", {arrayVector}, expectedArrayVector);
-  }
-  {
-    auto arrayVector = makeArrayVector<int64_t>({{1, 2, 3, 4}});
-    auto expectedArrayVector = makeArrayVector<int64_t>({{}});
-    VELOX_ASSERT_THROW(
-        testSlice(
-            "slice(C0, 1, -1)",
-            {arrayVector, arrayVector, expectedArrayVector},
-            expectedArrayVector),
-        "The value of length argument of slice() function should not be negative");
+};
+
+class SparkSliceTest : public SliceTest {
```

Comment:
Spark's tests should be put in the sparksql/tests folder.

Reply:
We have to reuse `commonTestCases` here, otherwise we could have duplicates in `sparksql/tests`(https://github.com/facebookincubator/velox/pull/9620#discussion_r1682194565), is that suitable?

Reply:
> It would be nice to refactor to avoid code duplication.

One method I can think of is to extract helper functions in a base class, and extends it in Presto and Spark tests. cc: @mbasmanova 

This is an example:
Base: https://github.com/facebookincubator/velox/blob/main/velox/functions/lib/aggregates/tests/SumTestBase.h

Presto: https://github.com/facebookincubator/velox/blob/main/velox/functions/prestosql/aggregates/tests/SumTest.cpp

Spark: https://github.com/facebookincubator/velox/blob/main/velox/functions/sparksql/aggregates/tests/SumAggregationTest.cpp

Reply:
Done